File: src/component_learning_py/component_learning_py/data_display_component.py
"""数据显示组件 —— 订阅滤波后的温度数据并打印"""
import rclpy
from rclpy.node import Node
from sensor_msgs.msg import Temperature


class DataDisplayComponent(Node):
    """数据显示组件"""

    def __init__(self, **kwargs):
        super().__init__('data_display', **kwargs)

        # 声明参数
        self.declare_parameter('input_topic', 'filtered_temperature')
        input_topic = self.get_parameter('input_topic').value

        self._count = 0
        self._sub = self.create_subscription(
            Temperature, input_topic, self._on_filtered_temperature, 10)

        self.get_logger().info(
            f'DataDisplayComponent initialized, sub={input_topic}'
        )

    def _on_filtered_temperature(self, msg):
        self._count += 1
        self.get_logger().info(
            f'[{self._count:4d}] Filtered temperature: '
            f'{msg.temperature:.2f} C (variance: {msg.variance:.4f})'
        )


def main(args=None):
    """独立运行入口"""
    rclpy.init(args=args)
    node = DataDisplayComponent()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: src/component_learning_cpp/src/data_display_component.cpp
#include "component_learning_cpp/data_display_component.hpp"
#include <rclcpp_components/register_node_macro.hpp>

/// 数据显示组件实现
DataDisplayComponent::DataDisplayComponent(const rclcpp::NodeOptions & options)
: Node("data_display", options)
{
  // 声明参数
  this->declare_parameter("input_topic", std::string("filtered_temperature"));

  std::string input_topic = this->get_parameter("input_topic").as_string();

  sub_ = this->create_subscription<sensor_msgs::msg::Temperature>(
    input_topic, 10,
    std::bind(&DataDisplayComponent::on_filtered_temperature, this, std::placeholders::_1));

  RCLCPP_INFO(
    this->get_logger(),
    "DataDisplayComponent initialized, sub=%s",
    input_topic.c_str());
}

void DataDisplayComponent::on_filtered_temperature(
  const sensor_msgs::msg::Temperature::SharedPtr msg)
{
  count_++;
  RCLCPP_INFO(
    this->get_logger(),
    "[%4zu] Filtered temperature: %.2f C (variance: %.4f)",
    count_, msg->temperature, msg->variance);
}

RCLCPP_COMPONENTS_REGISTER_NODE(DataDisplayComponent)
